101-sql-database-Import/azuredeploy.json

```diff
@@ -0,0 +1,122 @@
+      "metadata": {
+        "description": "The name of the SQL logical server."
+      }
+    },
+    "sqlDBName": {
+      "type": "string",
+      "defaultValue": "SampleDB",
+      "metadata": {
+        "description": "The name of the SQL Database."
+      }
+    },
+    "storageAccountType": {
+      "type": "string",
+      "allowedValues": ["LRS","ZRS","GRS"],
+      "defaultValue": "GRS"
+    },
+    "location": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]",
+      "metadata": {
+        "description": "Location for all resources."
+      }
+    },
+    "administratorLogin": {
+      "type": "string",
+      "metadata": {
+        "description": "The administrator username of the SQL logical server."
+      }
+    },
+    "administratorLoginPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "The administrator password of the SQL logical server."
+      }
+    },
+      "storageAccountKey": {
+      "type": "string",
+      "defaultValue":"",
+      "metadata": {
+        "description": "Specifies the key of the storage account where the BACPAC file is stored."
+      }
+    },
+    "bacpacUrl": {
+      "type": "string",
+      "defaultValue":"",
+      "metadata": {
+        "description": "Specifies the URL of the BACPAC file."
+      }
+    }
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Sql/servers",
+      "apiVersion": "2020-02-02-preview",
+      "name": "[parameters('serverName')]",
+      "location": "[parameters('location')]",
+      "properties": {
+        "administratorLogin": "[parameters('administratorLogin')]",
+        "administratorLoginPassword": "[parameters('administratorLoginPassword')]"
+      },
+      "resources": [
+                {
+                    "type": "firewallrules",
+                    "apiVersion": "2020-02-02-preview",
+                    "name": "AllowAllAzureIps",
+                    "location": "[parameters('location')]",
+                    "dependsOn": [
+                        "[parameters('serverName')]"
+                    ],
+                    "properties": {
+                        "startIpAddress": "0.0.0.0",
+                        "endIpAddress": "0.0.0.0"
+                    }
+               }
+        ,
+        {
+          "type": "databases",
+          "apiVersion": "2020-02-02-preview",
+          "name": "[parameters('sqlDBName')]",
+          "location": "[parameters('location')]",
+          "properties":
+          {
+            "storageAccountType": "[parameters('storageAccountType')]"
+          },
+          "sku": {
+            "name": "Standard",
+            "tier": "Standard"
+          },
+          "dependsOn": [
+            "[resourceId('Microsoft.Sql/servers', concat(parameters('serverName')))]"
+          ],
+          "resources": [
+                {
+                    "condition":"[not(empty(parameters('bacpacUrl')))]",
```

Comment:
this needs to be tested, you can check other samples in the repo for a bacpac you can use

Reply:
it was tested and is successful, this allows users to import to a database while we set the backup storage redundancy. 
this is not implemented yet for any of the import options (Portal, SQLPackage) this is why this is so important

Reply:
Not sure I follow re: NYI... if it's not implemented in the platform we should remove it from the sample.  If it is in the platform we want to test it in the pipeline.

Reply:
This template should be used by customers that need to import Azure SQL DB where policy enforced to use local redundant storage for backups. 
the regular import interface (using portal or command line tools) are not yet implementing this option. 
this is why this option is so important. 
here is a blog post I published about it: https://techcommunity.microsoft.com/t5/azure-database-support-blog/import-database-when-policy-quot-azure-sql-db-should-avoid-using/ba-p/1960857

Reply:
Thanks, that helps clarify... We're providing a sample for the import case, we just need to be able to test that code.  Here are the changes we need:

1) the import resource should not be conditional - if it is, this is just a template that creates a db and a server and there are plenty of samples that do that already
2) the import resource needs to be tested, there are a number of samples that do this already, you can look at those for how it's done...  Essentially you can use a sasUri for the bacpac and provide it with the sample.  You can provide the option to use a StorageKey, but CI will use the sas key.